tests/types/resolving/test_forward_references.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+
+import strawberry
+from strawberry.annotation import StrawberryAnnotation
+
+
+def test_forward_reference():
+    global ForwardClass
+
+    annotation = StrawberryAnnotation("ForwardClass", namespace=globals())
+
+    @strawberry.type
+    class ForwardClass:
+        backward: bool
+
+    resolved = annotation.resolve()
+
+    assert resolved is ForwardClass
+
+    del ForwardClass
+
+
+@pytest.mark.xfail(reason="Combining locals() and globals() strangely makes this fail")
```

Comment:
is this for another PR? :)

Reply:
Yes, I don't understand why it doesn't work, but it shouldn't affect normal end-users. If you want to take a look, go ahead :slightly_smiling_face: